### PR TITLE
Add JSON representation of Profile Preview Card info

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

To complete [RFC 222](https://github.com/forem/rfcs/pull/222) we need to be able to fetch via async JS some info about the user's profile. The content mimics the HTML representation, without the profile image which is already available on the client.

This PR only adds the endpoint, @aitchiss will later make use of it in the frontend

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/222

## QA Instructions, Screenshots, Recordings

1. start the app
2. Go to `http://localhost:3000/profile_preview_cards/:id` (`:id` is the ID of a user) to see the HTML representation
3. Go to `http://localhost:3000/profile_preview_cards/:id.json` to see the JSON representation
4. Make sure the hover on the profile preview cards still works

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

File: app/controllers/profile_preview_cards_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class ProfilePreviewCardsController < ApplicationController
```
Comment (rhymes) [inline app/controllers/profile_preview_cards_controller.rb:1]: I renamed the controller and paths to follow the rails convention of having controllers with plural names

File: app/controllers/profile_preview_cards_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  layout false
+
+  def show
+    @user = User.find(params[:id])
```
Comment (rhymes) [inline app/controllers/profile_preview_cards_controller.rb:5]: renamed `@actor` to `@user` as we only care about users here

File: app/javascript/packs/commentDropdowns.js [hunk 3]
```diff
@@ -63,7 +63,9 @@ const fetchMissingProfilePreviewCard = async (placeholderElement) => {
     jsCommentUserId: commentUserId,
     jsDropdownContentId: dropdownContentId,
   } = placeholderElement.dataset;
-  const response = await window.fetch(`/profile_preview_card/${commentUserId}`);
+  const response = await window.fetch(
+    `/profile_preview_cards/${commentUserId}`,
```
Comment (rhymes) [inline app/javascript/packs/commentDropdowns.js:67]: added the new path

File: app/views/profile_preview_cards/show.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<%= render "shared/profile_preview_card", actor: @user, id: nil %>
```
Comment (rhymes) [inline app/views/profile_preview_cards/show.html.erb:1]: nothing has changed, except `@user` instead of `@actor`

**Comment by aitchiss** [comment]
Gave this a quick test locally and it's looking awesome 🔥  The only thing we're missing that we need for the card is the user's colors - in the `html.erb` version of the preview card we call:

```rb
Color::CompareHex.new([user_colors(actor)[:bg], user_colors(actor)[:text]]).brightness(0.88)
```

Is there a way to extract the results of `user_colors()` into the json?

**Comment by rhymes** [comment]
@aitchiss sure, let me add that!

**Review by aitchiss — APPROVED** [review]
Looks good to me! 🙌

**Review by msarit — APPROVED** [review]
Great work by the wizard @rhymes 🧙🏾‍♂️  🚀

**Pull request merged**